### DISCUSSION TITLE
Switch to more recent c++ standard

### DISCUSSION THREAD

**Discussion by vifactor** [COVESA/dlt-viewer#501] [Ideas]
Currently, in CMakeLists it is explicitly stated that project is developed using c++11. Qt6 which is supported in the project requires c++17-capable compiler. All compilers that are used on CI by the project are like that. Is there any chance to make a switch to more recent standard, for example to use std::optional or other useful goodies?

**Comment by vifactor** [comment]
Related question about build systems: is there any particular reason why project maintains both qmake and cmake build systems @alexmucde? Starting from Qt6 cmake is the official way to go, while already in Qt5 for many years qmake is considered as almost deprecated.

**Comment by vifactor** [comment]
from now on it is possible to use c++17

**Comment by alexmucde** [comment]
@vifactor DLT Viewer was started as a qmake project very long time ago, over 10 years. Sometimes some people contributed a cmake configuration later. Since this time both make systems are supported. Now as cmake is verywhere supported, there is no real need anymore to support qmake anymore, but we need to get all missing parts running also with cmake first.
